File: problems/1. Two Sum/Solution.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Key is num, value is its index.
        Map<Integer, Integer> map = new HashMap<>();
        int[] r = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int currentNum = nums[i];
            if (map.containsKey(target - currentNum)) {
                r[0] = map.get(target - currentNum);
                r[1] = i;
                return r;
            } else {
                map.put(currentNum, i);
            }
        }
        return r;
    }
}

File: problems/1. Two Sum/Solution.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        buff_dict = {}
        for i in range(len(nums)):
            com = target - nums[i]
            if com in buff_dict:
                return [i, buff_dict[com]]
            else:
                buff_dict[nums[i]] = i
